### PR TITLE
Conditional compiler directive

### PR DESCRIPTION
**Pull request by minuscorp**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Solves  #17423

### CODE REVIEW

**Review by joshdholtz — COMMENTED** [review]
This looks great! Does this just need some testing for it to be ready to ship? 😊

**Comment by minuscorp** [comment]
Yeah, i wrote this from the ipad because i was sure of what was the issue, but I haven't been able of testing it. Also, I wasn't totally sure of this being a real issue, but as we do support Swift 4, this directive had to be change, which is against of the apple docs (irony: _weird doesn't it?_) so tomorrow I'll compile in swift 4 mode and see if it works just fine!

File: snapshot/lib/assets/SnapshotHelper.swift [hunk 1]
```diff
@@ -180,7 +180,11 @@ open class Snapshot: NSObject {
                 simulator = regex.stringByReplacingMatches(in: simulator, range: range, withTemplate: "")
 
                 let path = screenshotsDir.appendingPathComponent("\(simulator)-\(name).png")
-                try image.pngData()?.write(to: path, options: .atomic)
+                #if Swift(<5.0)
```
Comment (rogerluan) [inline snapshot/lib/assets/SnapshotHelper.swift:183]: >Unexpected platform condition (expected 'os', 'arch', or 'swift')

```suggestion
                #if swift(<5.0)
```

🎉

**Review by rogerluan — CHANGES_REQUESTED** [review]

**Review by rogerluan — APPROVED** [review]
LGTM! 🚀

**Pull request merged**